clients/v1_shell/autocomplete.go

```diff
@@ -0,0 +1,203 @@
+package v1shell
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/influxdata/go-prompt"
+)
+
+type SuggestNode struct {
+	Description  string
+	subsuggestFn func(string) (map[string]SuggestNode, string)
+}
+
+func (c *Client) suggestUse(remainder string) (map[string]SuggestNode, string) {
+	s := map[string]SuggestNode{}
+	for _, db := range c.Databases {
+		s["\""+db+"\""] = SuggestNode{Description: "Table Name"}
+	}
+	return s, remainder
+}
+
+func (c *Client) suggestDropMeasurement(remainder string) (map[string]SuggestNode, string) {
+	s := map[string]SuggestNode{}
+	if c.Database != "" && c.RetentionPolicy != "" {
+		for _, m := range c.Measurements {
+			s["\""+m+"\""] = SuggestNode{Description: fmt.Sprintf("Measurement on \"%s\".\"%s\"", c.Database, c.RetentionPolicy)}
+		}
+	}
+	return s, remainder
+}
+
+func (c *Client) suggestDelete(remainder string) (map[string]SuggestNode, string) {
+	s := map[string]SuggestNode{}
+	fromReg := regexp.MustCompile(`(?i)FROM(\s+)` + identRegex("from_clause") + `?$`)
+	matches := reSubMatchMap(fromReg, remainder)
+	if matches != nil {
+		if c.Database != "" && c.RetentionPolicy != "" {
+			for _, m := range c.Measurements {
+				s["\""+m+"\""] = SuggestNode{Description: fmt.Sprintf("Measurement on \"%s\".\"%s\"", c.Database, c.RetentionPolicy)}
+			}
+		}
+		return s, getIdentFromMatches(matches, "from_clause")
+	}
+	return s, remainder
+}
+
+func (c *Client) suggestSelect(remainder string) (map[string]SuggestNode, string) {
```

Comment:
It could be really cool to complete fields names eventually - not for this PR though. Needs some UX thought.